hera_cal/tests/test_redcal.py

```diff
@@ -1499,6 +1499,55 @@ def test_redcal_iteration(self):
+        # split up data into multiple time slices
+        # run redcal to produce metadatae
+        input_data = os.path.join(DATA_PATH, 'zen.2458098.43124.subband.uvh5')
+        uvd = io.HERAData(input_data)
+        uvd.read()
+        tmppath = tmpdir.strpath
+        # split into six different ten integration files
+        data_files = []
+        meta_files = []
+        times = np.unique(uvd.time_array)
+        for tnum in range(6):
+            uvd_slice = uvd.select(times=times[tnum * 10: (tnum + 1) * 10],
+                                   inplace=False)
+            t0 = times[tnum * 10]
+            output_file = os.path.join(tmppath, f'zen.{t0:.5f}.subband.uvh5')
+            uvd_slice.write_uvh5(output_file)
+            data_files.append(output_file)
+        # now run redcal on each
+        ant_metrics_file = os.path.join(DATA_PATH, 'test_input/zen.2458098.43124.HH.uv.ant_metrics.json')
+        for dfile in data_files:
+            cal = om.redcal_run(dfile, verbose=True, ant_z_thresh=1.8, add_to_history='testing',
+                                a_priori_ex_ants_yaml=os.path.join(DATA_PATH, 'test_input', 'a_priori_flags_sample.yaml'),
+                                iter0_prefix='.iter0', metrics_files=ant_metrics_file, clobber=True, ex_ants=[11, 50])
+            hd = io.HERAData(input_data)
+            meta_files.append(dfile.replace('.uvh5', '.redcal_meta.hdf5'))
+        # now run median
+        om.nightly_median_firstcal_delays(meta_files)
+        # now make sure all delays and polarity flips are equal to the median.
+        meta_files_median = [meta_file.replace('redcal_meta', 'redcal_meta.median_phases') for meta_file in meta_files]
+        for fnum, metafile in enumerate(meta_files_median):
+            fc_meta = io.read_redcal_meta(metafile)[0]
+            if fnum == 0:
+                median_dlys = {ant: fc_meta['dlys'][ant][0] for ant in fc_meta['dlys']}
+                median_polarity = {ant: fc_meta['polarity_flips'][ant][0] for ant in fc_meta['polarity_flips']}
+            else:
+                assert np.all([np.all(fc_meta['dlys'][ant] == median_dlys[ant]) for ant in fc_meta['dlys'] if np.isfinite(median_dlys[ant])])
+                assert np.all([np.all(fc_meta['polarity_flips'][ant] == median_polarity[ant]) for ant in fc_meta['polarity_flips'] if np.isfinite(median_polarity[ant])])
+
+        # now replace the redcal degeneracies.
+        omnifiles = [dfile.replace('.uvh5', '.omni.calfits') for dfile in data_files]
+        for omni_file, meta_file, meta_file_old in zip(omnifiles, meta_files_median, meta_files):
+            om.update_redcal_phase_degeneracy(omni_file, meta_file, meta_file_old,
+                                              output_file=omni_file.replace('.calfits', '.medphase.calfits'), clobber=True)
+
+        # check that files have been created
+        for ofile in omnifiles:
+            assert os.path.exists(omni_file.replace('.calfits', '.medphase.calfits'))
+
```

Comment:
Is there any other test we can do? Like, does doing this make the calibration solutions smoother in time? If so, can we quantify that and make an assertion about it?

Reply:
I think it's overkill to show whether or not the calibration is smoother. Anecdotally, I find that the differences in the solutions after replacing the degeneracies are quite small. I will document this above.